meteor/sofie-server/data-access/repositories/mongo/mongo-part-repository.ts

```diff
@@ -33,4 +35,22 @@ export class MongoPartRepository extends BaseMongoRepository implements PartRepo
 			})
 		)
 	}
+
+	public async deletePartsForSegment(segmentId: string): Promise<void> {
+		this.assertDatabaseConnection(this.deletePartsForSegment.name)
+		const parts: Part[] = await this.getParts(segmentId)
+
+		parts.forEach(async (part: Part) => this.pieceRepository.deletePiecesForPart(part.id))
```

Comment:
Refer to my comment on possibly avoiding the "Promise returned from forEach argument is ignored" issue in the `mongo-segment-repository.ts` file.

Reply:
If possible we want to use forEach loops, instead of for..of loops, if i have understood Lindved correctly.
That said i have changed it back to the for..of loop, to get rid of the warning.

Reply:
Another solution would also to use map and Promise.all, such that this becomes:
```typescript
await Promise.all(
  parts.map(async (part: Part) => this.pieceRepository.deletePiecesForPart(part.id))
)
```

By doing this we will also avoid getting unhandled promise rejections :smile:

Reply:
Suggestion applied.